modules/services/polybar.nix

```diff
@@ -132,18 +132,11 @@ in
       };
     };
 
-    home.activation.checkPolybar = dag.entryBefore [ "linkGeneration" ] ''
-      if ! cmp --quiet \
-          "${configFile}" \
-          "$HOME/.config/polybar/config"; then
-        polybarChanged=1
-      fi
-    '';
-
-    home.activation.applyPolybar = dag.entryAfter [ "reloadSystemD" ] ''
-      if [[ -v polybarChanged && -v DISPLAY ]]; then
+    # TODO: should be after reloadSystemD
```

Comment:
Actually, I believe this shouldn't be done through the `onChange` mechanism at all. Instead the systemd service file should contain a reference to the configuration file so that the regular systemd reload can detect the change.

So I would suggest reverting the change in this module for now and then I can set up a separate PR to use the systemd service file trick.